src/format/LemLibOdomGeneratorFormatV0_4.tsx

```diff
@@ -0,0 +1,267 @@
+import { euclideanRotation } from "../core/Coordinate";
+
+// observable class
+class GeneralConfigImpl implements GeneralConfig {
+  @IsPositive()
+  @Expose()
+  robotWidth: number = 12;
+  @IsPositive()
+  @Expose()
+  robotHeight: number = 12;
+  @IsBoolean()
+  @Expose()
+  robotIsHolonomic: boolean = false;
+  @IsBoolean()
+  @Expose()
+  showRobot: boolean = false;
+  @ValidateNumber(num => num > 0 && num <= 1000) // Don't use IsEnum
+  @Expose()
+  uol: UnitOfLength = UnitOfLength.Inch;
+  @IsPositive()
+  @Expose()
+  pointDensity: number = 2; // inches
+  @IsPositive()
+  @Expose()
+  controlMagnetDistance: number = 5 / 2.54;
+  @Type(() => FieldImageSignatureAndOrigin)
+  @ValidateNested()
+  @IsObject()
+  @Expose()
+  fieldImage: FieldImageSignatureAndOrigin<FieldImageOriginType> =
+    getDefaultBuiltInFieldImage().getSignatureAndOrigin();
+  @Expose()
+  chassisName: string = "chassis";
+  @Expose()
+  movementTimeout: number = 5000;
+  @Exclude()
+  private format_: LemLibOdomGeneratorFormatV0_1;
+
+  constructor(format: LemLibOdomGeneratorFormatV0_1) {
+    this.format_ = format;
+    makeAutoObservable(this);
+
+    reaction(
+      () => this.uol,
+      action((_: UnitOfLength, oldUOL: UnitOfLength) => {
+        convertGeneralConfigUOL(this, oldUOL);
+      })
+    );
+
+    intercept(this, "fieldImage", change => {
+      const { assetManager } = getAppStores();
+
+      if (assetManager.getAssetBySignature(change.newValue.signature) === undefined) {
+        change.newValue = getDefaultBuiltInFieldImage().getSignatureAndOrigin();
+      }
+
+      return change;
+    });
+  }
+
+  get format() {
+    return this.format_;
+  }
+
+  getConfigPanel() {
+    const { app } = getAppStores();
+    return (
+      <>
+        <Box className="panel-box">
+          <Typography sx={{ marginTop: "16px" }}>Export Settings</Typography>
+          <Box className="flex-editor-panel">
+          <ObserverInput
+            label="Chassis Name"
+            getValue={() => this.chassisName}
+            setValue={(value: string) => {
+              app.history.execute(
+                `Change chassis variable name`,
+                new UpdateProperties(this as any, { chassisName: value })
+              );
+            }}
+            isValidIntermediate={() => true}
+            isValidValue={(candidate: string) => candidate !== ""}
+            sx={{ marginTop: "16px" }}
+          />
+          <ObserverInput
+            label="Movement Timeout"
+            getValue={() => this.movementTimeout.toString() }
+            setValue={(value: string) => {
+              const parsedValue = parseInt(value)!;
+              app.history.execute(
+                `Change default movement timeout`,
+                new UpdateProperties(this as any, { movementTimeout: parsedValue })
+              );
+            }}
+            isValidIntermediate={() => true}
+            isValidValue={(candidate: string) => parseInt(candidate) !== undefined}
```

Comment:
1. Attribute `numeric` should be added to the `ObserverInput`.

2. Error occurs when calling `parseInt("")`, which is equal to `NaN`. Please check out `ControlAccordion` line 18 to learn more about how `ObserverInput` should handle number input. You better use `Tokens.Int.parse` to parse the value.